sklearn/metrics/classification.py

```diff
@@ -1035,6 +1000,17 @@ def precision_recall_fscore_support(y_true, y_pred, beta=1.0, labels=None,
                     raise ValueError("pos_label=%r is not a valid label: %r" %
                                      (pos_label, present_labels))
             labels = [pos_label]
+        else:
+            raise ValueError("Target is %s but average='binary'. Please "
+                             "choose another average setting. You may then "
+                             "use labels=[pos_label] to specify a single "
+                             "positive class." % y_type)
+    elif pos_label not in (None, 1):
+        warnings.warn("Note that pos_label (set to %r) is ignored when "
+                      "average != 'binary' (got %r). You may use "
+                      "labels=[pos_label] to specify a single positive class."
```

Comment:
if ``labels`` has a single entry, the ``average`` parameter doesn't matter, right?